securedrop/source_app/utils.py

```diff
@@ -32,7 +34,7 @@ def valid_codename(codename):
     return source is not None
 
 
-def generate_unique_codename(config):
+def generate_unique_codename(config: SDConfig) -> Optional[str]:
```

Comment:
On second look, thinking this should be `(config: SDConfig) -> str:`.

Reply:
yep `str` should be the return type here (the function will keep running until it finds a valid codename)